mlrun/data_types/infer.py

```diff
@@ -26,8 +26,14 @@ def upsert_entity(name, value_type):
         else:
             entities[name] = {"name": name, "value_type": value_type}
 
-    for column, series in df.items():
-        value_type = _get_column_type(series)
+    schema = pyarrow.Schema.from_pandas(df)
+    index_type = None
+    for i in range(len(schema)):
+        column = schema.names[i]
+        value_type = pa_type_to_value_type(schema.types[i])
+        if column in df.index.names:
+            index_type = value_type
```

Comment:
if i > 1 you will run over index_type value, if i am  not mistaken

Reply:
MultiIndex is not supported (before and after this PR).